src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -87,3 +119,303 @@ export const validateStorageSubmissionParams = celebrate({
     version: Joi.number().required(),
   }),
 })
+
+export const createFormsgReqBody: SharedSubmissionMiddlewareHandlerType =
+  async (req, res, next) => {
+    if (req.body.formsg) return res.send(new FormsgReqBodyExistsError())
+    else {
+      req.body.formsg = {}
+      return next()
+    }
+  }
+
+export const retrieveForm: SharedSubmissionMiddlewareHandlerType = async (
+  req,
+  res,
+  next,
+) => {
+  const { formId } = req.params
+
+  const logMeta = {
+    action: 'retrieveForm',
+    ...createReqMeta(req),
+    formId,
+  }
+
+  // Retrieve form
+  const formResult = await FormService.retrieveFullFormById(formId)
+  if (formResult.isErr()) {
+    logger.warn({
+      message: 'Failed to retrieve form from database',
+      meta: logMeta,
+      error: formResult.error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(formResult.error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  req.body.formsg.formDef = formResult.value
```

Comment:
should we store in res.locals? (as this was not actually part of the original request body)

Reply:
That was something I did originally (can check out to [d22f450](https://github.com/opengovsg/FormSG/pull/6595/commits/d22f450baad1001336308a95bded67ecbff1849a)) but I was discussing with @timotheeg about this, and he mentioned that the purpose for `res.locals` is for template rendering purposes back before single page applications were popular. As according to [Express' docs on res.locals](https://expressjs.com/en/api.html#res.locals):

> Use this property to set variables accessible in templates rendered with [res.render](https://expressjs.com/en/api.html#res.render). The variables set on res.locals are available within a single request-response cycle, and will not be shared between requests.

And though it is not part of the original request body, the intermediate contexts passed on can be seen as context based on the requests before the final response can be returned.

Reply:
It does technically work though 😂 just not very sure about the correctness / conventions

Reply:
thanks! sounds good to me, lets go with `req.body<...>` then :)

Reply:
⚠️ I'd recommend **not** injecting data in `req.body` either! `request.body` is populated from incoming data (from whatever body parser middleware we use), and there should be a clear distinction between incoming data, and server-generated, or server-retrieved data. 

Folks may have concerns on polluting the `req` object, my approach is always to create an app-specific namespace into `req`, maybe something like this:

```javascript
if (!req.formsg) req.formsg = {}

req.formsg.something = 'foobar'
```

On using `res.locals`, this link is the answer I aligned the most with (read the whole thing because there is specifically a discussion on how locals relates to rendering and is **not** the recommended approach).
https://stackoverflow.com/a/71487552/361295

expressjs in fact specifically updated their documentation on `res.locals` to make it clear that `res.locals` relates to the `res.render()` and it not meant to be the place to dump cross-middleware variables (what @LinHuiqing shared above)

Functionally all the approaches work of course, `req` and `res` are static objects available for the lifetime of a request, so you could shove stuff anywhere, but I strongly recommend keeping a good mental model of what things are:
* keep raw incoming data separate
* keep processed incoming data separate
* keep render-concern data separate
* keep request lifecycle data (cross-middleware contextual data that gets aggregated by one or more middleware) separate

I mentally model all that like this: say you retrieve the session data and the associated user from the sessionid cookie, that data is contextual properties of **the request**, not the response! Same thing if you retrieve full product data from the db on a request because you need it to make conditional decisions. Regardless of what the response will be, the product data is also a contextual property of **the request**.

Reply:
Ohh I see, thanks @timotheeg for taking the time to review this and dropping your comments! 
Hm okay let me see if I can create a `req.formsg` instead. I don't think there's a precedent in the code base yet 🤔

Reply:
Refactored so that `req.formsg` is used to pass context instead! And `req.body` is not modified throughout.